src/plugins/crypto/crypto.h

```diff
@@ -13,22 +13,49 @@
 #include <kdbplugin.h>
 #include <stdio.h>
 
-#include <crypto_internal.h>
-
-// methods for kdb
-int elektraCryptoOpen(Plugin *handle, Key *errorKey);
-int elektraCryptoClose(Plugin *handle, Key *errorKey);
-int elektraCryptoGet(Plugin *handle, KeySet *ks, Key *parentKey);
-int elektraCryptoSet(Plugin *handle, KeySet *ks, Key *parentKey);
-int elektraCryptoError(Plugin *handle, KeySet *ks, Key *parentKey);
-
-// provider-independent crypto functions
-int elektraCryptoInit(Key *errorKey);
-void elektraCryptoTeardown();
-int elektraCryptoHandleCreate(elektraCryptoHandle **handle, KeySet *config, Key *errorKey);
-void elektraCryptoHandleDestroy(elektraCryptoHandle *handle);
-int elektraCryptoEncrypt(elektraCryptoHandle *handle, Key *k, Key *errorKey);
-int elektraCryptoDecrypt(elektraCryptoHandle *handle, Key *k, Key *errorKey);
+enum ElektraCryptoHeaderFlags
+{
+	ELEKTRA_CRYPTO_FLAG_NONE = 0,
+	ELEKTRA_CRYPTO_FLAG_STRING = 1,
+	ELEKTRA_CRYPTO_FLAG_NULL = 2
+};
+
+#define ELEKTRA_CRYPTO_META_ENCRYPT ("crypto/encrypt")
+#define ELEKTRA_CRYPTO_META_ENCRYPTED ("crypto/encrypted")
```

Comment:
The docu in METADATA.ini also does not explain why there are two states needed. It would be rather unusual if this is actual needed, because in kdbGet() you already have the implicit knowledge that tagged keys are encrypted and need to be decrypted; and in kdbSet() you know its the other way round.

Why not use `crypto/encrypt` only, which means that keys are encrypted when persisted in files but obviously unencrypted (even though still tagged with `crypto/encrypt`) when handled to the user.

Two states would be needed, e.g. if you want to support to read unencrypted files and encrypt them before handing them to the user. Thats not very useful or is it? Maybe two states are useful if you failed to encrypt values and you did not want to fail kdbGet() completely? Is this the case?

Nevertheless, I would expect such a rationale in the documentation.